### PR TITLE
Closes #1302 - .yml files for setup of user and developer conda environment

### LINKED ISSUE #1302: Add yaml environment file to configure anaconda environment
Add a yaml file to allow for easy configuration of conda environment and dependency installation for using arkouda.

### PR DESCRIPTION
**Pull request by joshmarshall1**
This ticket (closes #1302):

Added two yaml files, one for standard configuration and one for developer configuration of conda environments. The packages included were the ones listed in `setup.py`.

These files successfully generated conda environments in testing that were able to run arkouda without errors, but we should find a way to test these in a completely clean environment without other previous installations or existing packages.

### CODE REVIEW

File: arkouda-env-dev.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  - pyfiglet
+  - versioneer
+  - matplotlib
+  - h5py
+  - pip
+  # Developer dependencies
+  - pexpect
+  - pytest
+  - pytest-env
+  - Sphinx
```
Comment (Ethan-DeBandi99) [inline arkouda-env-dev.yml:19]: This should be lowercase `sphinx`.

File: arkouda-env-dev.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  - sphinx-argparse
+  - sphinx-autoapi
+  - typed-ast
+  
+  - pip:
+    - typeguard==2.10.0
+    # Developer dependencies
+    - mypy==0.931
```
Comment (Ethan-DeBandi99) [inline arkouda-env-dev.yml:27]: According to `setup.py` this should be `mypy>=0.931`. I know there were issues when your version was higher. Can we track that down and determine which case is correct and set it? We can probably add a separate issue to track this.

**Review by pierce314159 — APPROVED** [review]
Not a YAML or env config expert but this looks good to me

**Review by Ethan-DeBandi99 — CHANGES_REQUESTED** [review]
Nothing major, just the one word that needs lowercase. The other comment should probably be addressed in its own PR as what is in both locations will function properly.

File: arkouda-env-dev.yml (hunk 1 above)
Comment (mhmerrill) [reply arkouda-env-dev.yml:19]: this is supposed to be uppercase I think.
https://github.com/Bears-R-Us/arkouda/blob/b9e53938bce78a34be0b6f1631525f5b1db95f52/setup.py#L154-L156

File: arkouda-env-dev.yml (hunk 2 above)
Comment (mhmerrill) [reply arkouda-env-dev.yml:27]: agree with this comment.
https://github.com/Bears-R-Us/arkouda/blob/b9e53938bce78a34be0b6f1631525f5b1db95f52/setup.py#L154-L156

**Review by mhmerrill — CHANGES_REQUESTED** [review]
see comments

**Comment by Ethan-DeBandi99** [comment]
> If we're opting to make parquet a requirement, it might be good to add the parquet dependencies to this .yml as well. @Ethan-DeBandi99 @pierce314159 @mhmerrill Thoughts?

@joshmarshall1 - we will be doing this in the future, but currently `pyarrow` and `Parquet` are optional. I agree when we switch to make these dependencies, we should add it to the `.yml`, but for now it should not be added.

File: arkouda-env-dev.yml (hunk 2 above)
Comment (joshmarshall1) [reply arkouda-env-dev.yml]: Updated to by `mypy>=0.931`
I will add an issue to track down potential version issue

**Review by Ethan-DeBandi99 — APPROVED** [review]
looks good.

**Review by pierce314159 — APPROVED** [review]

**Comment by Ethan-DeBandi99** [comment]
@mhmerrill - are there further changes you want made before this PR is ready?

**Review by mhmerrill — APPROVED** [review]

**Pull request merged**